clients/src/main/java/org/apache/kafka/clients/admin/DescribeUserScramCredentialsResult.java

```diff
@@ -0,0 +1,139 @@
+        KafkaFuture<Void> succeedsOnlyIfUsersFutureSucceeds = KafkaFuture.allOf(users());
+        return succeedsOnlyIfUsersFutureSucceeds.thenApply(void1 -> {
+            KafkaFuture<Void> succeedsOnlyIfAllDescriptionsSucceed = KafkaFuture.allOf(perUserFutures.values().toArray(
+                    new KafkaFuture[perUserFutures.size()]));
+            KafkaFuture<Map<String, UserScramCredentialsDescription>> mapFuture = succeedsOnlyIfAllDescriptionsSucceed.thenApply(void2 ->
+                perUserFutures.entrySet().stream().collect(Collectors.toMap(
+                    e -> e.getKey(),
+                    e -> valueFromFutureGuaranteedToSucceedAtThisPoint(e.getValue()))));
+            /* At this point it is only the users future that is guaranteed to have succeeded.
+             * We want to return the future to the map, but we have to return a map at this point.
+             * We need to dereference the future while propagating any exception.
+             */
+            return valueFromFuturePropagatingExceptionsAsUnchecked(mapFuture);
+        });
+    }
+
+    /**
+     *
+     * @return a future indicating the distinct users that were requested (either explicitly or implicitly via
+     * describe-all).  The future will not complete successfully if the user is not authorized to perform the describe
+     * operation; otherwise, it will complete successfully as long as the list of users with credentials can be
+     * successfully determined within some hard-coded timeout period.
+     */
+    public KafkaFuture<List<String>> users() {
+        return usersFuture;
+    }
+
+    /**
+     *
+     * @param userName the name of the user description being requested
+     * @return a future indicating the description results for the given user. The future will complete exceptionally if
+     * the future returned by {@link #users()} completes exceptionally.  If the given user does not exist in the list
+     * of requested users then the future will complete exceptionally with
+     * {@link org.apache.kafka.common.errors.ResourceNotFoundException}.
+     */
+    public KafkaFuture<UserScramCredentialsDescription> description(String userName) {
+        KafkaFuture<Void> succeedsOnlyIfUsersFutureSucceeds = KafkaFuture.allOf(usersFuture);
+        return succeedsOnlyIfUsersFutureSucceeds.thenApply(void1 -> {
+            // it is possible that there is no future for this user (for example, the original describe request was for
+            // users 1, 2, and 3 but this is looking for user 4), so explicitly take care of that case
+            KafkaFuture<UserScramCredentialsDescription> requestedUserFuture = perUserFutures.get(userName);
+            if (requestedUserFuture == null) {
+                throw new ResourceNotFoundException("No such user: " + userName);
+            }
+            KafkaFuture<Void> succeedsOnlyIfRequestedUserFutureSucceeds = KafkaFuture.allOf(requestedUserFuture);
```

Comment:
We don't need this extra future, right?  We can just use requestedUserFuture directly if we want to use that